### PR TITLE
Apply card view to pages

### LINKED ISSUE #822: Convert form page views to card view
As a user [all user types: volunteer, admin, supervisor] I want to click on a view/create/edit item in the menu sidebar,
so a new form route renders.

**How does it work now?**
Clicking on any view/create/edit opens a new page.

**How should it work?**
Clicking on any view/create/edit button so a new form route renders.

**Which pages / forms should this apply to?**
All, including:
- [ ] #837 View / Create / Edit `casa_case`
- [ ] #829 View / Create / Edit `case_contacts`
- [ ] #835 View / Create / Edit `volunteers`
- [ ] #841 View / Create / Edit `supervisors`
- [ ] #844 View / Create / Edit `admins`
- [ ] #832 Edit user profile
- [ ] #846  Edit CASA Org

**Permissions Changes Required?**
None

**Launch Plan**
Roll out to Admins and Supervisors first, then all users. 

**Acceptance Criteria**
Forms are accessible by the new blue sidebar menu.
Forms expand on the same page when opened.
Forms do not open in a new page.

Mockups:
[casa-new-contact.pdf](https://github.com/rubyforgood/casa/files/5270195/casa-new-contact.pdf)

### PR DESCRIPTION
**Pull request by ds29**
### What github issue is this PR for, if any?
Resolves #822, #837, #829, #835, #841, #844, #832, #846 

### What changed, and why?

Primary changes were to add card view wrapper around tables and forms. Also added more consistent "Back" navigation on nested routes to ensure the user can more logically navigate through the system.
### How will this affect user permissions?
- Volunteer permissions: No changes
- Supervisor permissions: No changes
- Admin permissions: No changes

### How is this tested? (please write tests!) 💖💪

Test by existing coverage and a lot of manual navigation through each area of the site.

### Screenshots please :)

I won't include all of them but here are a couple to give a sampling of the card view applied.

supervisors
case-contacts
volunteers

volunteer-cases

### Feelings gif (optional)
Yay

### CODE REVIEW

File: app/views/reports/index.html.erb [hunk 1]
```diff
@@ -1,46 +1,35 @@
-<%= link_to "Back", :back %>
+<h1>Report Categories</h1>
```
Comment (derricklannaman) [inline app/views/reports/index.html.erb:1]: @Dustin are we missing a link to back out of this page?

Comment (derricklannaman) [reply app/views/reports/index.html.erb:1]: Otherwise, LGTM

Comment (ds29) [reply app/views/reports/index.html.erb:1]: @derricklannaman I excluded it from any page that had a direct link in the sidebar.

Screenshot here for reference: reports

Comment (derricklannaman) [reply app/views/reports/index.html.erb:1]: Cool

**Comment by compwron** [comment]
You are invited to the CASA asynchronous retrospective-and-futurespective! Please take a look at https://bit.ly/casaretro (no login required) and add at least one note! Even if it's just "I was here and you are cool" or "omg y'all why do you keep messaging me" we want to hear it. <3
Next week Wednesday 18 November @ 6pm Pacific time we will Zoom about the board and talk about themes and takeaways. (Zoom link in our calendar at http://bit.ly/casacal or slack #casa channel https://rubyforgood.herokuapp.com/ )
Please ESPECIALLY add lots of notes if you won't be able to make it on Wednesday to talk!
* Note: the board will be monitored for spam; message me or any team lead if you see a problem
https://bit.ly/casaretro

**Pull request merged**